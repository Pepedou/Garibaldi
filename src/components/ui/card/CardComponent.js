import React, { Component } from "react";
import PropTypes from "prop-types";
import {
  Card,
  CardActions,
  CardMedia,
  CardTitle,
  CardText
} from "material-ui/Card";

export default class CardComponent extends Component {
  render() {
    return (
      <Card>
        <CardMedia
          overlay={
            <CardTitle
              title={this.props.overlayTitle}
              subtitle={this.props.overlaySubtitle}
            />
          }
        >
          <img src={this.props.cardImage} alt="" />
        </CardMedia>
        <CardTitle
          title={this.props.cardTitle}
          subtitle={this.props.cardSubtitle}
        />
        <CardText>{this.props.cardDescription}</CardText>
        <CardActions>{this.props.cardActions}</CardActions>
      </Card>
    );
  }
}

CardComponent.propTypes = {
  overlayTitle: PropTypes.any,
  overlaySubtitle: PropTypes.any,
  cardImage: PropTypes.string,
  cardTitle: PropTypes.any,
  cardSubtitle: PropTypes.any,
  cardDescription: PropTypes.any,
  cardActions: PropTypes.any
};
