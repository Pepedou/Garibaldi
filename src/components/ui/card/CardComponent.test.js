import CardComponent from "./CardComponent";
import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";

let props = {
  overlayTitle: "",
  overlaySubtitle: "",
  cardImage: "",
  cardTitle: "",
  cardSubtitle: "",
  cardDescription: "",
  cardActions: ""
};

it("CardComponent renders", () => {
  const tree = renderer
    .create(
      <MuiThemeProvider>
        <CardComponent {...props} />
      </MuiThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
