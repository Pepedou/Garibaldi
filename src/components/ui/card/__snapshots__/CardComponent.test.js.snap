// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent renders 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <div
      style={
        Object {
          "muiPrepared": true,
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        <img
          alt=""
          src=""
          style={
            Object {
              "maxWidth": "100%",
              "minWidth": "100%",
              "muiPrepared": true,
              "verticalAlign": "top",
              "width": "100%",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "muiPrepared": true,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "muiPrepared": true,
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "background": "rgba(0, 0, 0, 0.54)",
                "bottom": 0,
                "left": 0,
                "muiPrepared": true,
                "paddingTop": 8,
                "position": "absolute",
                "right": 0,
              }
            }
          >
            <div
              style={
                Object {
                  "muiPrepared": true,
                  "padding": 16,
                  "position": "relative",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "display": "block",
                    "fontSize": 24,
                    "lineHeight": "36px",
                    "muiPrepared": true,
                  }
                }
              >
                
              </span>
              <span
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 0.54)",
                    "display": "block",
                    "fontSize": 14,
                    "muiPrepared": true,
                  }
                }
              >
                
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
        }
      }
    >
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "block",
            "fontSize": 24,
            "lineHeight": "36px",
            "muiPrepared": true,
          }
        }
      >
        
      </span>
      <span
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "display": "block",
            "fontSize": 14,
            "muiPrepared": true,
          }
        }
      >
        
      </span>
    </div>
    <div
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontSize": 14,
          "muiPrepared": true,
          "padding": 16,
        }
      }
    >
      
    </div>
    <div
      style={
        Object {
          "muiPrepared": true,
          "padding": 8,
          "position": "relative",
        }
      }
    />
  </div>
</div>
`;
